### ISSUE TITLE
Test not using tmpdir fixture

### ISSUE THREAD

**Issue by richardjgowers** [MDAnalysis/mdanalysis#2099]
Running the testsuite generates this file:

`testsuite/MDAnalysisTests/data/rmsfit_adk_dims.dcd`

Probably because a test is not using the `tmpdir` fixture.

This test should use the tmpdir fixture and not leave a trace

**Comment by orbeckst** [comment]
Yes, I have always found that one annoying... just didn't have the time to track it down.

**Comment by PicoCentauri** [comment]
I just checked the corresponding test. The test checks the default file name, which would not be tested properly for an explicit path using the tmpdir feature. https://github.com/MDAnalysis/mdanalysis/blob/25b27e13be4079e9cdb6a156d2120005e7bc9ae8/testsuite/MDAnalysisTests/analysis/test_align.py#L187-L190

The easiest (but dirty) fix would be to remove the file after the test.

**Comment by richardjgowers** [comment]
@PicoCentauri the problem with deleting after the test is it assumes the test exits cleanly.  It looks like it just needs to be ran in a `tmpdir`

**Comment by PicoCentauri** [comment]
I tried this first but this won't help since the filename attributes points in the directory where the trajectory is stored:
https://github.com/MDAnalysis/mdanalysis/blob/8918d42d1da0c653f41295fb7a131d7c04d46578/package/MDAnalysis/analysis/align.py#L622-L624

One could change this by just using the actual file name and ignore its path.

**Comment by richardjgowers** [comment]
huh, that's a pretty weird design.  Ok a pytest fixture which does nothing but cleanup would fix this, something like

```python
@pytest.fixture
def nuke_file():
    try:
        yield
    finally:
        os.remove(filename)

**Comment by PicoCentauri** [comment]
hmm looks good, I could write a simple patch. But maybe it's better to change the function itself? I think ist more convenient if the file is written to the current directory.

**Comment by richardjgowers** [comment]
@PicoCentauri yeah a patch would be very welcome!

Not sure about changing the behaviour, we might have to assume someone out there is using this weird behaviour as part of a huge workflow and we can't just break their hypothetical work etc...

**Comment by orbeckst** [comment]
We could deprecate the old behavior and change it for 1.0. writing to a specific path or current dir is more sane than what we currently have. 



--
Oliver Beckstein
email: orbeckst@gmail.com
